sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/SecretsImpl.java

```diff
@@ -110,7 +123,51 @@ public Secret getByNameAndVersion(String name, String version) {
 
     @Override
     public Mono<Secret> getByNameAndVersionAsync(final String name, final String version) {
-        return inner.getSecret(name, version).map(this::wrapModel);
+        Objects.requireNonNull(name);
+        return (version == null ? inner.getSecret(name) : inner.getSecret(name, version)).map(this::wrapModel);
+    }
+
+    @Override
+    public Secret enableByNameAndVersion(String name, String version) {
+        return enableByNameAndVersionAsync(name, version).block();
+    }
+
+    @Override
+    public Mono<Secret> enableByNameAndVersionAsync(String name, String version) {
+        Objects.requireNonNull(name);
+        return enableByNameAndVersionAsync(name, version, true);
+    }
+
+    @Override
+    public void disableByNameAndVersion(String name, String version) {
+        disableByNameAndVersionAsync(name, version).block();
+    }
+
+    @Override
+    public Mono<Void> disableByNameAndVersionAsync(String name, String version) {
+        Objects.requireNonNull(name);
+        return enableByNameAndVersionAsync(name, version, false).then();
+    }
+
+    private Mono<Secret> enableByNameAndVersionAsync(String name, String version, boolean enabled) {
```

Comment:
Is it better use something like `makeSecretEnabledOrDisabled`? Not just named with `enable`

Reply:
It is private method so I am a bit relax on the naming. I will try to figure out a better name.

Reply:
Changed to `updateSecretEnableDisableAsync`